File: code/3777/3777砖块.py
def check(s, c):
    res = []
    bk = list(s)
    for i in range(n - 1):
        if bk[i] != c:
            res.append(i)
            bk[i] = 'B' if bk[i] == 'W' else 'W'
            bk[i + 1] = 'B' if bk[i + 1] == 'W' else 'W'

    if bk[-1] != c:
        return False

    print(len(res))
    for x in res:
        print(x + 1, end=' ')
    if res:
        print()

    return True


if __name__ == "__main__":
    T = int(input())
    for _ in range(T):
        n = int(input())
        s = input()
        if not check(s, 'W') and not check(s, 'B'):
            print(-1)

File: code/3777/3777砖块.cpp
#include <algorithm>
#include <cstring>
#include <iostream>
#include <vector>
using namespace std;

int T, n;
string s;

void update(char &c) {
  if (c == 'B')
    c = 'W';
  else
    c = 'B';
}

bool check(char c) {
  vector<int> res;
  string str = s;
  for (int i = 0; i + 1 < n; i++) {
    if (str[i] != c) {
      res.push_back(i);
      update(str[i]);
      update(str[i + 1]);
    }
  }

  if (str.back() != c)
    return false;

  cout << res.size() << endl;
  for (int x : res)
    cout << x + 1 << " ";
  if (res.size())
    cout << endl;

  return true;
}

int main() {
  cin >> T;
  while (T--) {
    cin >> n >> s;
    if (!check('B') && !check('W'))
      puts("-1");
  }

  return 0;
}
